package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/reboot/0100-mesh-vpn.lua

```diff
@@ -35,6 +37,15 @@ elseif has_fastd then
 	else
 		msg = site_i18n._translate('gluon-config-mode:novpn')
 	end
+elseif has_wireguard then
+	local wireguard_enabled = uci:get_bool("wireguard", "mesh_vpn", "enabled")
+	if wireguard_enabled then
+		local secret = uci:get("wireguard", "mesh_vpn", "secret")
+		pubkey = util.trim(util.exec("/usr/bin/wg pubkey <<< " .. secret))
```

Comment:
Key generation should also happen when the config mode is not being used. In that case they key needs to be generated on first use.

Reply:
ash doesn't support `<<<`.

How about `uci get -q wireguard.mesh_vpn.secret | wg pubkey`?

Reply:
<<< is incorrect here anyways as the content is actually a file.